src/interactivity/win32/screenInfoUiaProvider.cpp

```diff
@@ -97,26 +97,24 @@ void ScreenInfoUiaProvider::ChangeViewport(const SMALL_RECT NewWindow)
     _pUiaParent->ChangeViewport(NewWindow);
 }
 
-HRESULT ScreenInfoUiaProvider::GetSelectionRanges(_In_ IRawElementProviderSimple* pProvider, const std::wstring_view wordDelimiters, _Out_ std::deque<ComPtr<UiaTextRangeBase>>& result)
+HRESULT ScreenInfoUiaProvider::GetSelectionRange(_In_ IRawElementProviderSimple* pProvider, const std::wstring_view wordDelimiters, _COM_Outptr_result_maybenull_ Microsoft::Console::Types::UiaTextRangeBase** ppUtr)
 {
-    try
+    RETURN_HR_IF_NULL(E_INVALIDARG, ppUtr);
+    *ppUtr = nullptr;
+    UiaTextRange* result = nullptr;
+    if (_pData->IsSelectionActive())

```

Comment:
From what I can find, it seems like the only place this is used is in `ScreenInfoUiaProviderBase::GetSelection()`, which also does a check for `_pData->IsSelectionActive()` before calling this function. I'm thinking this might be an unneeded check unless I'm not seeing where else it's used? Thoughts?